x-pack/plugins/ml/public/components/job_selector/custom_selection_table/custom_selection_table.js

```diff
@@ -0,0 +1,390 @@
+    firstItemIndex: 0,
+    lastItemIndex: 1
+  });
+  const [query, setQuery] = useState(EuiSearchBar.Query.MATCH_ALL);
+  const [error, setError] = useState(null); // eslint-disable-line
+
+  useEffect(() => {
+    setCurrentItems(items);
+    handleQueryChange({ query: query });
+  }, [items]); // eslint-disable-line
+
+  // When changes to selected ids made via badge removal - update selection in the table accordingly
+  useEffect(() => {
+    setItemIdToSelectedMap(getCurrentlySelectedItemIdsMap());
+  }, [selectedIds]); // eslint-disable-line
+
+  useEffect(() => {
+    const tablePager = new Pager(currentItems.length, JOBS_PER_PAGE);
+    setPagerSettings({
+      itemsPerPage: JOBS_PER_PAGE,
+      firstItemIndex: tablePager.getFirstItemIndex(),
+      lastItemIndex: tablePager.getLastItemIndex()
+    });
+    setPager(tablePager);
+  }, [currentItems]);
+
+  function getCurrentlySelectedItemIdsMap() {
+    const selectedIdsMap = { 'all': false };
+    selectedIds.forEach(id => { selectedIdsMap[id] = true; });
+    return selectedIdsMap;
+  }
+
+  function handleSingleSelectionTableChange(itemId) {
+    onTableChange([itemId]);
+  }
+
+  function handleTableChange({ isSelected, itemId }) {
+    const allIds = Object.getOwnPropertyNames(itemIdToSelectedMap);
+    let currentSelected = allIds;
+
+    if (itemId !== 'all') {
+      currentSelected = allIds.filter((id) =>
+        itemIdToSelectedMap[id] === true && id !== itemId);
+
+      if (isSelected === true) {
+        currentSelected.push(itemId);
+      }
+    } else {
+      if (isSelected === false) {
+        currentSelected = [];
+      } else {
+        // grab all id's
+        currentSelected = currentItems.map((item) => item.id);
+      }
+    }
+
+    onTableChange(currentSelected);
+  }
+
+  function handleChangeItemsPerPage(itemsPerPage) {
+    pager.setItemsPerPage(itemsPerPage);
+    setPagerSettings({
+      ...pagerSettings,
+      itemsPerPage,
+      firstItemIndex: pager.getFirstItemIndex(),
+      lastItemIndex: pager.getLastItemIndex()
+    });
+  }
+
+  function handlePageChange(pageIndex) {
+    pager.goToPageIndex(pageIndex);
+    setPagerSettings({
+      ...pagerSettings,
+      firstItemIndex: pager.getFirstItemIndex(),
+      lastItemIndex: pager.getLastItemIndex()
+    });
+  }
+
+  function handleQueryChange({ query: incomingQuery, error: newError }) {
+    if (newError) {
+      setError(newError);
+    } else {
+      const queriedItems = EuiSearchBar.Query.execute(incomingQuery, items, { defaultFields: filterDefaultFields });
+      setError(null);
+      setCurrentItems(queriedItems);
+      setQuery(incomingQuery);
+    }
+  }
+
+  function isItemSelected(itemId) {
+    return itemIdToSelectedMap[itemId] === true;
+  }
+
+  function areAllItemsSelected() {
+    const indexOfUnselectedItem = currentItems.findIndex(item => !isItemSelected(item.id));
+    return indexOfUnselectedItem === -1;
+  }
+
+  function renderSelectAll(mobile) {
```

Comment:
Nice to have: This could become a `<SelectAll .. />` stateless functional component and moved out of this component.